### PR TITLE
Add check for number of app responding to Intent.

### PR DESCRIPTION
**Pull request by mmosemmle**

### CODE REVIEW

File: pdk/src/main/java/com/pinterest/android/pdk/PDKClient.java [hunk 1]
```diff
@@ -157,8 +158,25 @@ public void logout() {
         saveAccessToken(null);
         saveScopes(null);
     }
+
+    public static boolean checkAppNumber(Context context, String clientId) {
```
Comment (NightlyNexus) [inline pdk/src/main/java/com/pinterest/android/pdk/PDKClient.java]: don't make this public?

File: pdk/src/main/java/com/pinterest/android/pdk/PDKClient.java [hunk 2]
```diff
@@ -157,8 +158,25 @@ public void logout() {
...
+        Intent testIntent = new Intent(Intent.ACTION_VIEW);
+        testIntent.setData(Uri.parse("pdk" + clientId + "://"));
+        PackageManager pm = context.getPackageManager();
+        List<ResolveInfo> activities = pm.queryIntentActivities(testIntent, 0);
+        if (activities.size() == 0) {
+            throw new IllegalStateException("No app responding to the Intent, please check the manifest to make sure the intent filter is set up properly.");
```
Comment (NightlyNexus) [inline pdk/src/main/java/com/pinterest/android/pdk/PDKClient.java]: if our app is not installed, i don't think we want to crash the calling app, do we? just fail the login call? or, am i wrong, and the calling app was supposed to check earlier?

File: pdk/src/main/java/com/pinterest/android/pdk/PDKClient.java [hunk 3]
```diff
@@ -157,8 +158,25 @@ public void logout() {
...
+        }
+        if (activities.size() != 1) return false;
+        return true;
+    }
+
     public void login (final Context context, final List<String> permissions, final PDKCallback callback) {
         _authCallback = callback;
+        if (!checkAppNumber(_context, _clientId)) {
+            if (callback != null) callback.onFailure(new PDKException("Unable to login because multiple apps are responding to the login request."));
```
Comment (NightlyNexus) [inline pdk/src/main/java/com/pinterest/android/pdk/PDKClient.java]: nit: `log in` as the verb, i think. `login` as an adjective and noun.

File: pdk/src/main/java/com/pinterest/android/pdk/PDKClient.java [hunk 4]
```diff
@@ -157,8 +158,25 @@ public void logout() {
...
+        if (activities.size() == 0) {
+            throw new IllegalStateException("No app responding to the Intent, please check the manifest to make sure the intent filter is set up properly.");
+        }
+        if (activities.size() != 1) return false;
+        return true;
+    }
+
     public void login (final Context context, final List<String> permissions, final PDKCallback callback) {
         _authCallback = callback;
+        if (!checkAppNumber(_context, _clientId)) {
```
Comment (NightlyNexus) [inline pdk/src/main/java/com/pinterest/android/pdk/PDKClient.java]: nit: use local `context` parameter?

File: pdk/src/main/java/com/pinterest/android/pdk/PDKClient.java (hunk 2 above)
Comment (mmosemmle) [reply pdk/src/main/java/com/pinterest/android/pdk/PDKClient.java]: This is actually checking whether calling app is responding to the Intent (This Intent is supposed to give control back to the calling app in the end of the OAuth process). If the calling app does not respond, then the appId in the manifest probably doesn't match up with the one coded in the app. (or some other reasons where the Intent Filter is not correct in the manifest) In production code, this should not happen.

**Comment by mmosemmle** [comment]
Thanks. Addressed the comments.

File: pdk/src/main/java/com/pinterest/android/pdk/PDKClient.java (hunk 2 above)
Comment (NightlyNexus) [reply pdk/src/main/java/com/pinterest/android/pdk/PDKClient.java]: oh, gotcha! right.

**Comment by NightlyNexus** [comment]
Thanks!

**Pull request merged**